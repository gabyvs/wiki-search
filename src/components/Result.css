.result {
  margin-bottom: 20px;
  color: white;
  border-radius: 2px;
  padding: 10px;
  width: 100%;
}

.result:hover {
  background-color: #434343;
}

.result-title {
  margin-bottom: 4px;
}

.result-title-link {
  color: #434343;
  text-decoration: none;
  font-size: 13px;
}

.result:hover .result-title-link {
  color: white;
}

.result-snippet {
  color: #c2c2c2;
  font-size: 13px;
}