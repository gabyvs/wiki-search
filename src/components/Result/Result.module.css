.result {
  margin-bottom: 20px;
  color: white;
  border-radius: 2px;
  padding: 10px;
  width: 100%;
}

.result:hover {
  background-color: #434343;
}

.title {
  margin-bottom: 4px;
}

.link {
  color: #434343;
  text-decoration: none;
  font-size: 13px;
}

.result:hover .link {
  color: white;
}

.snippet {
  color: #c2c2c2;
  font-size: 13px;
}