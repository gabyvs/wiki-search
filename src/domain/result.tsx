export interface IResult {
  title: string,
  snippet: string
}
