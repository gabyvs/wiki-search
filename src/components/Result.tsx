import React      from 'react';
import { IResult } from '../domain/result';
import './Result.css';

const Result = (props: IResult) => (
  <div className="result">
    <h3 className="result-title">
      <a className="result-title-link"
         href={props.url}
         target="_blank"
         rel="noopener">{props.title}</a>
    </h3>
    <div className="result-snippet">{props.snippet}</div>
    <a className="result-title-link"
       href={props.url}
       target="_blank"
       rel="noopener">{props.url}</a>
  </div>
);

export default Result;