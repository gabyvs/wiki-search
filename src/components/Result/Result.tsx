import React      from 'react';
import { IResult } from '../../domain/result';
import styles from './Result.module.css';

const Result = (props: IResult) => {
  const url = encodeURI(`https://en.wikipedia.org/wiki/${props.title}`);
  return (
    <div className={styles.result}>
      <h3 className={styles.title}>
        <a className={styles.link}
           href={url}
           target="_blank"
           rel="noopener">{props.title}</a>
      </h3>
      <div className={styles.snippet}>{props.snippet}</div>
      <a className={styles.link}
         href={url}
         target="_blank"
         rel="noopener">{url}</a>
    </div>
  );
};

export default Result;