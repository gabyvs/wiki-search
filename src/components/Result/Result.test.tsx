import React        from 'react';
import {
  shallow,
  ShallowWrapper }  from 'enzyme';
import Result       from './Result';
import { IResult }  from '../../domain/result';

describe('Result', () => {
  let result: ShallowWrapper;
  let props: IResult;
  let url: string;

  beforeEach(() => {
    props = {
      title: 'A title for the result',
      snippet: 'This would be a description for the result.'
    };
    url = encodeURI(`https://en.wikipedia.org/wiki/${props.title}`)
    result = shallow(<Result {...props} />)
  });

  it('renders without crashing', () => {});

  it('renders correctly', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders links with the passed url', () => {
    expect(result.find(`[href="${url}"]`)).toHaveLength(2);
  });

  it('renders title', () => {
    expect(result.find('a').at(0).text()).toBe(props.title);
  });

  it('renders url', () => {
    expect(result.find('a').at(1).text()).toBe(url);
  });

  it('renders snippet', () => {
    expect(result.find('.snippet').text()).toBe(props.snippet);
  });
});