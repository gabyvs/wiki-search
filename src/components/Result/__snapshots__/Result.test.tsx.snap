// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders correctly 1`] = `
<div
  className="result"
>
  <h3
    className="title"
  >
    <a
      className="link"
      href="https://en.wikipedia.org/wiki/A%20title%20for%20the%20result"
      rel="noopener"
      target="_blank"
    >
      A title for the result
    </a>
  </h3>
  <div
    className="snippet"
  >
    This would be a description for the result.
  </div>
  <a
    className="link"
    href="https://en.wikipedia.org/wiki/A%20title%20for%20the%20result"
    rel="noopener"
    target="_blank"
  >
    https://en.wikipedia.org/wiki/A%20title%20for%20the%20result
  </a>
</div>
`;
